sceptre-example/config/lambda-stack-bootstrapless.yaml

```diff
@@ -0,0 +1,55 @@
+# This is an example of deploying the CDK stack using the "bootstrapless" deployment_type. This is
+# the more "Sceptre-like" way to configure your stacks. You can "wire together" other stacks into
+# this one using resolvers to populate values in the bootstrapless_config, parameters, and
+# sceptre_user_data. By default, the profile or role doing the deployments will also be used to push
+# the file and image assets.
+template:
+  type: cdk
+  # The path is always within your project's templates/ directory.
+  path: lambda_stack.py
+  deployment_type: bootstrapless
+  # bootstrapless_config are the snake_cased arguments passed to the cdk-bootstrapless-synthesizer
+  # for definitions of possible parameters, see the API docs here:
+  # https://github.com/aws-samples/cdk-bootstrapless-synthesizer/blob/main/API.md
+  bootstrapless_config:
+    # You can use !stack_attr to reference other stack attributes that happen
+    # to be set with resolvers to chain the resolver value. It makes sense to use the
+    # same bucket as Sceptre uses for its template uploads for your file assets.
+    file_asset_bucket_name: !stack_attr template_bucket_name
```

Comment:
doesn't `file_asset_bucket_name` depend on assets_bucket.yaml and `image_asset_repository_name` depend on ecr.yaml?  if so do we need to define sceptre `dependencies` in this config?

Reply:
Actually, no. Using `!stack_attr` just references a different attribute/resolver already on this stack. In this case, it's referencing the `template_bucket_name`, which is defined on the `config.yaml` and thus carried over to every stack in the project. Since that config is set with `!stack_output`, it automatically makes this stack dependent upon the template bucket stack. `image_asset_repository_name` also `!stack_output` which automatically adds a dependency. In other words, you don't need to use an explicit dependency if you're using `!stack_output`, which both of these are doing (both directly and indirectly via !stack_attr).

Reply:
I _will_ take a look at this example, though, and see if there are a few comments I can make that make that more obvious for readers.

Reply:
ohh yes, i forgot that `!stack_output` implicitly sets up stack dependencies.